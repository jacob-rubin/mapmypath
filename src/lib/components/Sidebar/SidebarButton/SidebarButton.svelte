<script lang="ts">
	import MdiKeyboardArrowLeft from '~icons/mdi/keyboard-arrow-left';
	import MdiKeyboardArrowRight from '~icons/mdi/keyboard-arrow-right';

	interface Props {
		isOpen: boolean;
		onClick: () => void;
	}

	let { isOpen, onClick }: Props = $props();
</script>

<div
	data-testid={'tooltip'}
	class="tooltip tooltip-right flex"
	data-tip={isOpen ? 'Collapse' : 'Expand'}
>
	<button
		class="h-10 rounded-l-none rounded-r-lg border-none bg-neutral-content"
		onclick={onClick}
	>
		{#if isOpen}
			<MdiKeyboardArrowLeft data-testid={'collapse'} class="size-6" />
		{:else}
			<MdiKeyboardArrowRight data-testid={'expand'} class="size-6" />
		{/if}
	</button>
</div>
