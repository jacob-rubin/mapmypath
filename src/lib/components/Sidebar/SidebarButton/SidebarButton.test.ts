import { cleanup, render } from '@testing-library/svelte';
import { afterEach, describe, expect, it, vi } from 'vitest';
import SidebarButton from './SidebarButton.svelte';

describe('SidebarButton', async () => {
	afterEach(() => {
		cleanup();
	});

	it('matches the snapshot', async () => {
		const screen = render(SidebarButton, {
			isOpen: false,
			onClick: () => {}
		});

		expect(screen.container.innerHTML).toMatchSnapshot();
	});

	it('shows collapse button when isOpen is true', async () => {
		const screen = render(SidebarButton, {
			isOpen: true,
			onClick: () => {}
		});

		expect(screen.getByTestId('collapse')).toBeTruthy();
	});

	it('shows expand button when isOpen is true', async () => {
		const screen = render(SidebarButton, {
			isOpen: false,
			onClick: () => {}
		});

		expect(screen.getByTestId('expand')).toBeTruthy();
	});

	it('triggers the onclick callback when clicked', async () => {
		const onClick = vi.fn();
		const screen = render(SidebarButton, {
			isOpen: false,
			onClick: onClick
		});

		screen.getByRole('button').click();
		expect(onClick).toHaveBeenCalledOnce();
	});
});
