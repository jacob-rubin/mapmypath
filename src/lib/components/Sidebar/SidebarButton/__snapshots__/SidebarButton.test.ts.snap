// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SidebarButton > matches the snapshot 1`] = `"<div class="tooltip tooltip-right flex" data-testid="tooltip" data-tip="Expand"><button class="h-10 rounded-l-none rounded-r-lg border-none bg-neutral-content"><svg viewBox="0 0 24 24" width="1.2em" height="1.2em" data-testid="expand" class="size-6"><path fill="currentColor" d="M8.59 16.58L13.17 12L8.59 7.41L10 6l6 6l-6 6z"></path></svg><!----></button></div>"`;
